public/pages/Correlations/containers/CreateCorrelationRule.tsx

```diff
@@ -906,6 +906,12 @@ export const CreateCorrelationRule: React.FC<CreateCorrelationRuleProps> = (
             props.resetForm();
           }
 
+          useEffect(() => {
+            if (trigger?.id) {
```

Comment:
Is the `trigger.id` expected to change anytime? If not then instead of this effect we can simply do this in the initial state for `useState`